src/main/java/com/google/firebase/database/connection/NettyWebSocketClient.java

```diff
@@ -57,38 +60,51 @@
   private final ChannelHandler channelHandler;
   private final ExecutorService executorService;
   private final EventLoopGroup group;
+  private final boolean isSecure;
 
   private Channel channel;
 
   NettyWebSocketClient(
       URI uri, String userAgent, ThreadFactory threadFactory,
-      WebsocketConnection.WSClientEventHandler eventHandler) {
+      WebsocketConnection.WSClientEventHandler eventHandler, boolean isSecure) {
```

Comment:
Had to make this change to pass in `isSecure` attach `SSLContext` to the `WebSocketConnection` only if it is true. Earlier, all WS connections were assumed to be secure.

Reply:
Nit: Please make this the second argument. It makes for much more readable code if a consumer passes an inline lamdba into `NettyWebSocketClient`.